scripts/haveno_deps.sh

```diff
@@ -62,21 +60,34 @@ check_bitcoin() {
 
 # Download Monero bins
 dw_monero() {
+
+    extract_monero() {
+        echo "-> extracting monerod and monero-wallet-rpc from archive" && \
+        tar -xzf "monero-bins-haveno-${platform}.tar.gz" && \
+        chmod +x {monerod,monero-wallet-rpc} || exit 1
+    }
+
     if is_mac; then
         platform="mac"
     else
         platform="linux"
     fi
 
-    if [ -f monero-bins-haveno-${platform}.tar.gz ]; then
-        check_monero
+    if [ -f "monero-bins-haveno-${platform}.tar.gz" ] && [ -f "monerod"  ] && [ -f "monero-wallet-rpc" ]; then
```

Comment:
The binaries should be downloaded and extracted if the archive is wrong or missing, regardless of the extracted binaries, so we should remove monerod and monero-wallet-rpc from this conditional.